pkg/virt-controller/services/template.go

```diff
@@ -1180,24 +1180,9 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, t
 	// If an SELinux type was specified, use that--otherwise don't set an SELinux type
 	selinuxType := t.clusterConfig.GetSELinuxLauncherType()
 	if selinuxType != "" {
-		pod.Spec.SecurityContext.SELinuxOptions = &k8sv1.SELinuxOptions{Type: selinuxType}
-		// By setting an SELinux option on the virt-launcher pod, we trigger this:
-		// https://github.com/kubernetes/kubernetes/issues/90759
-		// Since the compute container needs to be able to communicate with the rest of the pod,
-		//   we loop over all the containers and remove their SELinux categories.
-		for i := range pod.Spec.Containers {
-			container := &pod.Spec.Containers[i]
-			if container.Name != "compute" {
-				if container.SecurityContext == nil {
-					container.SecurityContext = &k8sv1.SecurityContext{}
-				}
-				if container.SecurityContext.SELinuxOptions == nil {
-					container.SecurityContext.SELinuxOptions = &k8sv1.SELinuxOptions{}
-				}
-				container.SecurityContext.SELinuxOptions.Type = selinuxType
-				container.SecurityContext.SELinuxOptions.Level = "s0"
-			}
-		}
+		// more info on https://github.com/kubernetes/kubernetes/issues/90759
+		// all containers on a pod must have the same SELinux context
+		alignPodMultiCategorySecurity(&pod, selinuxType)
```

Comment:
Why don't we use a proper function comment instead of leaving the comment here? Also this might be a sign that the name of the method does not express the intent well enough, so as to make the comment redundant?

Reply:
I'm just pointing the reader to the full story.

Plus,  I think 'align pod multi category security' is a good name to someone that knows selinux internals... Do you think that is going into too much detail ? 

Would `alignPodSecurityContext` be better ? Even if we choose that name, I'd prefer to leave the k8s issue link.

Reply:
Sure, we can leave the link! What I was trying to say is that we should convert this rather into a proper function doc. See my proposal below.

Reply:
I find it weird to do document private functions (especially ones that are used only once). This might be me. 

I find more value in having them near the caller - but I see your point; it pollutes the code. I just think it does a better job of informing the user that the code should be frowned upon and more info is found at the link if she/he are up to follow this particular rabbit hole.

I guess a trade-off here is impossible. Let me think this through a bit more to see if I figure an intermediate way out.

Reply:
Moved the comment to the implementation of the function; you've convinced me that whoever is interested in the full story will just follow through to the implementation.